ee/packages/pdf-worker/src/strategies/ChatTranscript.ts

```diff
@@ -10,8 +12,41 @@ export class ChatTranscript implements IStrategy {
 		return !previous || !moment(current.ts).tz(timezone).isSame(previous.ts, 'day');
 	}
 
-	private parserMessages(messages: PDFMessage[], dateFormat: string, timeAndDateFormat: string, timezone: string): PDFMessage[] {
-		return messages.map((message, index, arr) => {
+	private async parseSystemMessage(messages: PDFMessage[]): Promise<PDFMessage[]> {
+		return Promise.all(
+			messages.map(async (message: PDFMessage) => {
+				if (!message.t) {
+					return message;
+				}
+
+				const systemMessageDefinition = getSystemMessage(message.t);
+
+				if (!systemMessageDefinition) {
+					return message;
+				}
```

Comment:
Same here, you can filter the ones we have definitions for and avoid pushing a promise that will return inmediately (for cases where systemMessageDefinition is undefined)